### PR TITLE
Buffer messages per stream

### PR DESCRIPTION
**Pull request by cosimon**
# Description of change
Buffer messages separately based on the stream. The target will still flush messages based on the total number of messages, but it will not flush on switching to a new stream.

# QA steps
 - [x] automated tests passing
 - [ ] manual qa steps passing (list below)

# Risks
High - This code touches every record message emitted by the target. If there is a bug in here it would affect every tap deployed after this PR is merged.

# Rollback steps
 - revert this branch

### CODE REVIEW

File: target_stitch/__init__.py [hunk 1]
```diff
@@ -599,24 +607,26 @@ def handle_line(self, line):
...
                 message.bookmark_properties)
 
         elif isinstance(message, (singer.RecordMessage, singer.ActivateVersionMessage)):
-            if self.messages and (
-                    message.stream != self.messages[0].stream or
-                    message.version != self.messages[0].version):
+            current_stream = message.stream
+            if self.messages[current_stream] and (
+                    message.version != self.messages[current_stream][0].version):
                 self.flush()
```
Comment (KAllan357) [inline target_stitch/__init__.py:621]: This used to be "flush when the stream changes or when the version changes, but is now just when the version changes. Could we document this behavior some with a comment.

File: target_stitch/__init__.py [hunk 2]
```diff
@@ -613,7 +613,7 @@ def handle_line(self, line):
                 self.contains_activate_version[current_stream] = True
 
             num_bytes = self.buffer_size_bytes
-            num_messages = len(self.messages[current_stream])
+            num_messages = sum((len(messages) for messages in self.messages.values()))
```
Comment (dmosorast) [inline target_stitch/__init__.py:629]: Does a test cover this case? Just to be extra sure.

File: target_stitch/__init__.py [hunk 3]
```diff
@@ -608,11 +608,11 @@ def handle_line(self, line):
                 self.flush()
 
             self.messages[current_stream].append(message)
-            self.buffer_size_bytes += len(line)
+            self.buffer_size_bytes[current_stream] = self.buffer_size_bytes.get(current_stream, 0) + len(line)
             if isinstance(message, singer.ActivateVersionMessage):
                 self.contains_activate_version[current_stream] = True
 
-            num_bytes = self.buffer_size_bytes
+            num_bytes = sum(self.buffer_size_bytes.values())
```
Comment (dmosorast) [inline target_stitch/__init__.py:628]: Does a test cover this case? (It should, but want to double check)

Comment (dmosorast) [reply target_stitch/__init__.py:628]: Good that it's still flushing based on total bytes

File: target_stitch/__init__.py (hunk 2 above)
Comment (cosimon) [reply target_stitch/__init__.py:629]: Just updated the test to check this condition

File: target_stitch/__init__.py (hunk 1 above)
Comment (cosimon) [reply target_stitch/__init__.py:623]: Added a comment above the relevant code. Do you want this documented somewhere else too?

File: target_stitch/__init__.py (hunk 3 above)
Comment (cosimon) [reply target_stitch/__init__.py:630]: Just added a test for this.

**Pull request closed without merging**